cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/MetadataDataset.java

```diff
@@ -124,9 +124,8 @@ public void setProperty(Id.NamespacedId targetId, String key, String value) {
    * @param tags the tags to set
    */
   private void setTags(Id.NamespacedId targetId, String ... tags) {
-    BusinessMetadataRecord tagsRecord =
-      new BusinessMetadataRecord(targetId, TAGS_KEY, Joiner.on(TAGS_SEPARATOR).join(tags));
-    setBusinessMetadata(tagsRecord, MetadataType.TAG);
+    MetadataEntry tagsEntry = new MetadataEntry(targetId, TAGS_KEY, Joiner.on(TAGS_SEPARATOR).join(tags));
```

Comment:
Not related to this PR but a lot of these static of this class can be static. Lets change them.